website/content/tutorials/3-constraints-expect/_index.md

```diff
@@ -0,0 +1,1511 @@
+        id: vendor3
+        name: Massive Memory Corp
+        website: https://example.com/massive-memory-corp/        
+{{< /highlight >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+We can use the samples above to simulate a scenario where a developer or their tool define an invalid memory module configuration. The table below identifies constraints and their evaluation against the new sample. We can observe the processor should report on specific constraint errors for some fields, but not others.
+
+| Constraint ID             | Evaluated Target                                                  | Evaluated Test            | Result |
+|---------------------------|-------------------------------------------------------------------|---------------------------|--------|
+| memory-divisible-two      | `<!-- 1st memory module -->`<br/>`<byte-size>8589934591</byte-size>` |  `(8589934591 mod 2) = 0`    | false  |
+|                           | `<!-- 2nd memory module -->`<br/>`<byte-size>8589934592</byte-size>` |  `(8589934592 mod 2) = 0`    | true   |
+| memory-divisible-megabyte | `<!-- 1st memory module -->`<br/>`<byte-size>8589934591</byte-size>` |  `(8589934591 mod 1024) = 0` | false  |
+|                           | `<!-- 2nd memory module -->`<br/>`<byte-size>8589934592</byte-size>` |  `(8589934592 mod 1024) = 0` | true   |
+| memory-same-byte-size     | Sequence of <br/> (`<memory>...<product-name>... (Module 1)</product-name><byte-size>8589934591</byte-size></memory>`, <br/>`<memory>...<product-name>... (Module 2)</product-name><byte-size>8589934592</byte-size></memory>`) | `(sum((8589934592, 8589934592)) mod 8589934592) = 0)` |  true |
+
+## Deprecation warnings with `expect`
+
+With our use of the `expect` constraint for logical value testing, our stakeholders were very pleased we can develop and test small model enhancements incrementally and quickly. However, the adoption of a standard information model and data formats have surfaced some problems and challenges they were unaware of, so they need our help. An important group in their consortium identified a risk around using `byte-size` with binary base-two number for memory or storage capacity. The use of [binary prefixes and units](https://en.wikipedia.org/wiki/Binary_prefix) are ambiguous in the new model. Some vendors use binary base-two sizes or decimal base-ten sizes, but this field in the model makes tooling inconsistent and document instances unclear. A majority in the consortium decided updates to documentation (sourced from our Metaschema modules) are not enough. There must be a way to identify size, its unit prefix (binary or decimal), and optionally a unit so developers and software do not need to compute the full byte size. Our stakeholders are very concerned because the consortium also wants to allow the old approach in our information model and mark this method as deprecated. The intent will be to remove after twelve to twenty four months.
```

Comment:
```suggestion
With our use of the `expect` constraint for logical value testing, our stakeholders were very pleased we can develop and test small model enhancements incrementally and quickly. However, the adoption of a standard information model and data formats have surfaced some problems and challenges they were unaware of, so they need our help. An important group in their consortium identified a risk around using `byte-size` with binary base-two number for memory or storage capacity. The use of [binary prefixes and units](https://en.wikipedia.org/wiki/Binary_prefix) are ambiguous in the new model. Some vendors use binary base-two sizes or decimal base-ten sizes, but this field in the model makes tooling inconsistent and document instances unclear. A majority in the consortium decided updates to documentation (sourced from our Metaschema modules) are not enough. There must be a way to identify size, its unit prefix (binary or decimal), and optionally a unit so developers and software do not need to compute the full byte size. Our stakeholders are very concerned because the consortium also wants to allow the old approach in our information model and mark this method as deprecated. The intent will be to remove the deprecated approach after twelve to twenty four months.
```